### PR TITLE
Migrate to  azure-sdk-for-go 

### PR DESCRIPTION
**Pull request by neolynx**
Fixes https://github.com/aptly-dev/aptly/issues/1290

## Description of the Change

- migrate to new azure sdk

## Checklist

- [x] FIXMEs, MD5 upload, ...

### CODE REVIEW

File: azure/public.go [hunk 1]
```diff
@@ -114,9 +115,9 @@ func (storage *PublishedStorage) LinkFromPool(publishedPrefix, publishedRelPath,
 	sourcePath string, sourceChecksums utils.ChecksumInfo, force bool) error {
 
 	relFilePath := filepath.Join(publishedRelPath, fileName)
-	// prefixRelFilePath := filepath.Join(publishedPrefix, relFilePath)
-	// FIXME: check how to integrate publishedPrefix:
-	poolPath := storage.az.blobPath(fileName)
+	prefixRelFilePath := filepath.Join(publishedPrefix, relFilePath)
+	// FIXME: check how to integrate publishedPrefix: done?
```
Comment (neolynx) [inline azure/public.go:119]: use storage.az.blobPath

File: azure/public.go [hunk 2]
```diff
@@ -236,12 +246,12 @@ func (storage *PublishedStorage) internalCopyOrMoveBlob(src, dst string, metadat
...
 // RenameFile renames (moves) file
 func (storage *PublishedStorage) RenameFile(oldName, newName string) error {
-	return storage.internalCopyOrMoveBlob(oldName, newName, nil, true /* move */)
+	return storage.internalCopyOrMoveBlob(oldName, newName, "", true /* move */)
 }
 
 // SymLink creates a copy of src file and adds link information as meta data
 func (storage *PublishedStorage) SymLink(src string, dst string) error {
-	return storage.internalCopyOrMoveBlob(src, dst, azblob.Metadata{"SymLink": src}, false /* move */)
+	return storage.internalCopyOrMoveBlob(src, dst, "SymLink", false /* do not remove src */)
```
Comment (neolynx) [inline azure/public.go:254]: fixme

**Review by iofq — CHANGES_REQUESTED** [review]
Need to update `debian/control` with the new azure go package

**Comment by neolynx** [comment]
> Need to update `debian/control` with the new azure go package

I tried that, unfortunately the azure sdk package is lacking some modules in the latest version on debian, so this has to wait.
I am also synchronizing with the official debian packaging, and will ask if the missing go modules (mainly azure, swaggo, ..) could be provided for trixie.
The github pipeline is not building with debian dependencies, so the debian/control does not matter there...

**Comment by iofq** [comment]
> > Need to update `debian/control` with the new azure go package
> 
> I tried that, unfortunately the azure sdk package is lacking some modules in the latest version on debian, so this has to wait. I am also synchronizing with the official debian packaging, and will ask if the missing go modules (mainly azure, swaggo, ..) could be provided for trixie. The github pipeline is not building with debian dependencies, so the debian/control does not matter there...

gotcha so this must just be for building from source (deb-src)?

**Review by iofq — APPROVED** [review]

**Comment by neolynx** [comment]
> > > Need to update `debian/control` with the new azure go package
> > 
> > 
> > I tried that, unfortunately the azure sdk package is lacking some modules in the latest version on debian, so this has to wait. I am also synchronizing with the official debian packaging, and will ask if the missing go modules (mainly azure, swaggo, ..) could be provided for trixie. The github pipeline is not building with debian dependencies, so the debian/control does not matter there...
> 
> gotcha so this must just be for building from source (deb-src)?

yes, this is just for debian native build, with debian only dependencies, like molior aptly and debian official builds...

**Pull request merged**